### PR TITLE
SafelinksSingleUser: New integration

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
A new Safelinks integration that uses the EXO V1 module instead of EXO V2 as in the other integration of the product.
Supported auth method: `device-code` flow (rather than the certificate flow - delegated permissions and not applications ones)

## Minimum version of Cortex XSOAR
- [X] 6.0.0

## Does it break backward compatibility?
   - [X] No

### CODE REVIEW

**Comment by Bargenish** [comment]
@yaron-libman Please review only the following:

- [ ] [Detailed Description](https://github.com/demisto/content/pull/16778/files#diff-1a6c6dbb775deb6ce06a1079ed24682a2f9923c9edcf3799e83b2ec24ad73570)
- [ ] [Release Notes](https://github.com/demisto/content/pull/16778/files#diff-88a9f450c23c247789594601e243659b249c288c80393bc0ae94eef85c6e6f3b)
- [ ] New added command - `o365-defender-safelinks-detail-report-get`: https://github.com/demisto/content/pull/16778/files#diff-08a7552cba02721e7d65d98c13f9db79c1a0a09eacc9f53afb50e85a66a45172R924-R958

File: Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.ps1 [hunk 1]
```diff
@@ -0,0 +1,1142 @@
...
+        User Principal Name (UPN) is the name of a system user in an email address format.
+
+        .PARAMETER password
+        Password is filled only if authentication method is basic auth.
+
+        .PARAMETER bearer_token
+        Valid bearer token value.
+
+        .EXAMPLE proxy
+        Wheter to user system proxy configuration or not.
```
Comment (yaakovi) [inline Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.ps1:125]: not used?

File: Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.ps1 [hunk 2]
```diff
@@ -0,0 +1,1142 @@
...
+            .LINK
+            https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-configurable-token-lifetimes#configurable-token-lifetime-properties-after-the-retirement
+        #>
+    }
+
+    [bool]IsAccessTokenExpired(){
+        if (!$this.access_token){
+            return $true
+        }
+        $current_time = [int][double]::Parse((Get-Date -UFormat %s)) - 30
```
Comment (yaakovi) [inline Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.ps1:405]: 30 seconds?
move the comment here

File: Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.yml [hunk 3]
```diff
@@ -0,0 +1,964 @@
...
+  name: url
+  defaultvalue: https://outlook.office365.com
+  type: 0
+  required: true
+- display: Email / User Principal Name (Required)
+  displaypassword: Password (Required for Basic authentication only)
+  name: credentials
+  type: 9
+  required: false
+- display: Trust any certificate (not secure)
```
Comment (yaakovi) [inline Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.yml:16]: no proxy support?

File: Packs/Microsoft365Defender/ReleaseNotes/3_0_0.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### New: O365 Defender SafeLinks - Single User
```
Comment (yaakovi) [inline Packs/Microsoft365Defender/ReleaseNotes/3_0_0.md:3]: shouldn't it be in a new pack like the rest of the `Single User` integrations?

**Review by yaakovi — COMMENTED** [review]
💯

**Comment by julieschwartz18** [comment]
Doc review complete for the following. @Bargenish - please confirm.
1. New added command - o365-defender-safelinks-detail-report-get in the O365DefenderSafeLinksSingleUser.yml file. Updated descriptions - please review.
2. Release Notes - edited the existing content, let me know if it is updated.
3.  Detailed Description - edited - please review.

File: Packs/Microsoft365Defender/ReleaseNotes/3_0_0.md (hunk 4 above)
Comment (Bargenish) [reply Packs/Microsoft365Defender/ReleaseNotes/3_0_0.md:2]: discussed in person.

File: Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.yml [hunk 5]
```diff
@@ -908,51 +908,51 @@ script:
           required: false
           isArray: false
         - name: page
-          description: Specifies the page number of the results you want to view. Valid input is a number between 1 and 1000. The default value is 1.
+          description: Sets the number of pages of results you want to view. Valid input is a number between 1 and 1000.
```
Comment (Bargenish) [inline Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.yml:911]: @julieschwartz18 this is inaccurate. its is not the number of pages but the page number.

File: Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.ps1 (hunk 2 above)
Comment (Bargenish) [reply Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinksSingleUser/O365DefenderSafeLinksSingleUser.ps1:405]: Done

**Review by yaakovi — APPROVED** [review]

**Pull request merged**